.buildkite/scripts/steps/checks/no_codeowners.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common/util.sh
+
+# logic is different for PRs
+if [ "$GITHUB_PR_TARGET_BRANCH" != "" ]; then
+  # ignore prs targetting main
+  if [ "$GITHUB_PR_TARGET_BRANCH" != "main" ]; then
+    # try to find a version matching the target branch
+    version=$(jq '.versions[] | select(.branch == env.GITHUB_PR_TARGET_BRANCH) | .version' versions.json)
+    # if the target branch represents a tracked version then run the check
+    if [ "$version" != "" ]; then
+      echo "--- removing codeowners file"
+      rm .github/CODEOWNERS || true
+      check_for_changed_files 'rm .github/CODEOWNERS' true
+    fi
+  fi
+# when we're running on-merge or something on a non-pr
+elif [ "$BUILDKITE_BRANCH" != "main" ]; then
+  # try to find a version matching the branch being built
+  version=$(jq '.versions[] | select(.branch == env.BUILDKITE_BRANCH) | .version' versions.json)
+  # if the branch being built represents a tracked version then run the check
+  if [ "$version" != "" ]; then
+    echo "--- removing codeowners file"
+    rm .github/CODEOWNERS || true
+    check_for_changed_files 'rm .github/CODEOWNERS' true
+  fi
+fi
```

Comment:
@brianseeders This logic is a lot more complicated, maybe you can help me golf it a bit, but based on my testing it seems to work great

```sh
$ GITHUB_PR_TARGET_BRANCH="8.2" ./.buildkite/scripts/steps/checks/no_codeowners.sh
--- removing codeowners file
$ GITHUB_PR_TARGET_BRANCH="feature/foo" ./.buildkite/scripts/steps/checks/no_codeowners.sh
$ GITHUB_PR_TARGET_BRANCH="8.2" ./.buildkite/scripts/steps/checks/no_codeowners.sh
--- removing codeowners file
$ GITHUB_PR_TARGET_BRANCH="main" ./.buildkite/scripts/steps/checks/no_codeowners.sh
$ GITHUB_PR_TARGET_BRANCH="" BUILDKITE_BRANCH="main" ./.buildkite/scripts/steps/checks/no_codeowners.sh
$ GITHUB_PR_TARGET_BRANCH="" BUILDKITE_BRANCH="8.0" ./.buildkite/scripts/steps/checks/no_codeowners.sh
$ GITHUB_PR_TARGET_BRANCH="" BUILDKITE_BRANCH="8.2" ./.buildkite/scripts/steps/checks/no_codeowners.sh
--- removing codeowners file
$ GITHUB_PR_TARGET_BRANCH="" BUILDKITE_BRANCH="feature/foo" ./.buildkite/scripts/steps/checks/no_codeowners.sh
```

Reply:
Oh, now that I think about it... `8.3` won't exist in `versions.json` when we branch main to create that branch, so it won't trigger this logic 🤦

Reply:
We could pull `versions.json` from main and hope that `versions.json` gets updated before `8.3` is branched...